core/services/ocr2/plugins/ocr2keeper/log_provider.go

```diff
@@ -106,7 +178,120 @@ func (c *LogProvider) PerformLogs(ctx context.Context) ([]plugintypes.PerformLog
 	return vals, nil
 }
 
-func (c *LogProvider) unmarshalLogs(logs []logpoller.Log) ([]performed, error) {
+func (c *LogProvider) StaleReportLogs(ctx context.Context) ([]plugintypes.StaleReportLog, error) {
```

Comment:
don't really like the repetition of logic in this function, maybe we can make it more elegant with generics as a followup